core/banking/transfers_common.go

```diff
@@ -160,12 +166,17 @@ func (e *Engine) processTransfer(
 
 	// ensure the party have enough funds for both the
 	// amount and the fee for the transfer
-	feeTransfer, discount, err := e.makeFeeTransferForFundsTransfer(ctx, assetType, amount, from, fromAcc, to)
+	feeTransfer, discount, err := e.makeFeeTransferForFundsTransfer(ctx, assetType, amount, from, fromAcc, fromDerivedKey, to)
 	if err != nil {
 		return nil, fmt.Errorf("could not pay the fee for transfer: %w", err)
 	}
 	feeTransferAccountType := []types.AccountType{fromAcc}
 
+	// transfer from sub account to owners general account
+	if fromDerivedKey != nil {
+		from = *fromDerivedKey
+	}
+
 	fromTransfer, toTransfer := e.makeTransfers(from, to, assetType.ID, "", toMarket, amount, &transferID)
```

Comment:
Where do we check that derived key is "owned" by the submitter of the transaction?

Reply:
I am still working on it. The PR is not done. But almost there!